### PR TITLE
kyoto-tycoon: fix deps, patches & test

### PR DESCRIPTION
**Pull request by gromgit**
* Reduce `lua` to build dep, add `pkg-config`
* Add `zlib` dependency per upstream docs
* Substitute inline patch with already-existing `formula-patches` version
* Add ephemeral ports patch to support Homebrew testing (not required otherwise)
* Add proper test
* Add license

Followup to #65993.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/kyoto-tycoon.rb [hunk 1]
```diff
@@ -12,10 +12,21 @@ class KyotoTycoon < Formula
...
+  depends_on "zlib"
 
-  patch :DATA
+  # Build patch (submitted upstream to info@fallabs.com)
+  patch do
+    url "https://raw.githubusercontent.com/Homebrew/formula-patches/955ce09/kyoto-tycoon/0.9.56.patch"
+    sha256 "7a5efe02a38e3f5c96fd5faa81d91bdd2c1d2ffeb8c3af52878af4a2eab3d830"
+  end
+
+  # Homebrew-specific patch to support testing with ephemeral ports (often returned by `free_port`)
```
Comment (SMillerDev) [inline Formula/kyoto-tycoon.rb:25]: Is there any reason why it would not be useful to submit this upstream?

File: Formula/kyoto-tycoon.rb [hunk 2]
```diff
@@ -24,18 +35,29 @@ def install
...
+    pid = fork do
+      exec bin/"ktserver", "-port", port.to_s, "-scr", testpath/"test.lua"
+    end
+    sleep 5
+
+    assert_match "Homebrew\tCool", shell_output("#{bin}/ktremotemgr script -port #{port} echo Homebrew Cool 2>&1")
+  ensure
+    Process.kill("TERM", pid)
+    Process.wait(pid)
+  end
```
Comment (SMillerDev) [inline Formula/kyoto-tycoon.rb:62]: This should be handled by Homebrew.

File: Formula/kyoto-tycoon.rb (hunk 1 above)
Comment (gromgit) [reply Formula/kyoto-tycoon.rb:25]: My bad. Mail to upstream sent.

File: Formula/kyoto-tycoon.rb (hunk 2 above)
Comment (gromgit) [reply Formula/kyoto-tycoon.rb:62]: Just to confirm: Homebrew auto-kills all `fork`ed processes, so I can remove the `ensure` block?

Comment (SMillerDev) [reply Formula/kyoto-tycoon.rb:62]: It _should_ kill all forked processes.

File: Formula/kyoto-tycoon.rb [hunk 3]
```diff
@@ -24,18 +37,26 @@ def install
...
-@@ -271,7 +271,7 @@
-       if (!logf("prepare", "started to open temporary databases under %s", tmppath.c_str()))
-         err = true;
-       stime = kc::time();
--      uint32_t pid = getpid() & kc::UINT16MAX;
-+      uint32_t pid = kc::getpid() & kc::UINT16MAX;
-       uint32_t tid = kc::Thread::hash() & kc::UINT16MAX;
-       uint32_t ts = kc::time() * 1000;
-       for (size_t i = 0; i < dbnum_; i++) {
+    pid = fork do
```
Comment (SMillerDev) [inline Formula/kyoto-tycoon.rb:55]:
```suggestion
    fork do
```

**Review by SMillerDev — APPROVED** [review]

**Pull request closed without merging**